[native] Add Linux Memory Manager (#22997)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/properties.rst

* **Default value:** ``0.25``

  See description for ``shared-arbitrator.memory-pool-min-free-capacity``

> LinuxMemoryChecker Properties

Reviewer: The following memory properties don't have to be LinuxMemoryChecker specific?  Can we just call it Memory Pushback or Memory Checker properties?

Reviewer 2: Changing it to be `Memory Checker Properties`.

> -----------------------------

Reviewer 3: Please align the `---` with the previous line.

Reviewer 2: Done

> Currently there is a LinuxMemoryChecker that extends from PeriodicMemoryChecker and is used for Linux systems only. 

Reviewer 4: ```suggestion
The LinuxMemoryChecker extends from PeriodicMemoryChecker and is used for Linux systems only. 
```

The LinuxMemoryChecker can be enabled by setting the CMake flag ``PRESTO_MEMORY_CHECKER_TYPE=LINUX_MEMORY_CHECKER``. 
The following properties for PeriodicMemoryChecker are as follows:

> ``systemMemPushbackEnabled``

Reviewer 3: Are these the actual names of the config? The config values doesn't use came case.

They are called `system-mem-pushback-enabled` and so on. Please fix.

Reviewer 2: Changing the names to match with https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/common/Configs.h#L275

Reviewer 2: ```
system-mem-pushback-enabled
system-mem-limit-gb
system-mem-shrink-gb
```

* **Default value:** ``0.25``

  See description for ``shared-arbitrator.memory-pool-min-free-capacity``

Memory Checker Properties
-------------------------

The LinuxMemoryChecker extends from PeriodicMemoryChecker and is used for Linux systems only. 
The LinuxMemoryChecker can be enabled by setting the CMake flag ``PRESTO_MEMORY_CHECKER_TYPE=LINUX_MEMORY_CHECKER``. 
The following properties for PeriodicMemoryChecker are as follows:

``system-mem-pushback-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``boolean``
* **Default value:** ``false``

If set to ``true``, starts memory limit checker to trigger memory pushback when
server is under low memory pressure.

``system-mem-limit-gb``
^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``

> * **Default value:** ``0``

Reviewer 3: This is not 0 either.
```
NUM_PROP(kSystemMemLimitGb, 55),
```

> Specifies the system memory limit that triggers the memory pushback or heap dump if

Reviewer 3: Maybe mention this only has an effect if `system-mem-pushback-enabled` is enabled. Same as done with `system-mem-shrink-gb`.

the server memory usage is beyond this limit. A value of zero means no limit is set.

``system-mem-shrink-gb``
^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``

> * **Default value:** ``0``

Reviewer 3: Actually, the default is
```
NUM_PROP(kSystemMemShrinkGb, 8),
```
